gateway/tests/tests/tests.rs

```diff
@@ -129,20 +102,21 @@ async fn test_gateway_authentication() -> Result<()> {
         amount: bitcoin::Amount::from_sat(100),
         address: bitcoin.get_new_address(),
     };
+    test_auth(&gw_password, |pw| client_ref.withdraw(pw, payload.clone())).await?;
+
+    task_group.shutdown_join_all().await
+}
+
+async fn test_auth<Fut>(gw_password: &str, func: impl Fn(String) -> Fut) -> Result<()>
+where
+    Fut: Future<Output = Result<Response, Error>>,
+{
     assert_eq!(
-        client
-            .withdraw("withdraw".to_string(), payload.clone())
-            .await?
-            .status(),
-        401
-    );
-    assert_ne!(
-        client
-            .withdraw(gw_password.clone(), payload,)
-            .await?
-            .status(),
+        // use random password here
+        func("foobar123456789".to_string()).await?.status(),
         401
     );
+    assert_ne!(func(gw_password.to_string()).await?.status(), 401);
```

Comment:
Maybe this should be eq `200` just to be more explicit.

Reply:
That actually confused me too, but then it felt like a unrelated change. Let me add a commit to fix it.

Reply:
200 would be testing auth, and all the other logic of the call.

Reply:
`get_balance` fails with a 500 error, leaving it to @okjodom to fix if possible. Maybe this test is actually just testing that authentication does or doesn't fail and is intended to error internally.